lms/djangoapps/completion_api/tests/test_views.py

```diff
@@ -373,3 +374,70 @@ def test_unauthenticated(self):
         self.client.force_authenticate(None)
         response = self.client.post(self.update_url, {'completion': 1})
         self.assertEqual(response.status_code, 401)
+
+
+@ddt.ddt
+class CompletionMobileViewTestCase(SharedModuleStoreTestCase):
+    """
+    Test that the CompletionView with mobile courses.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(CompletionMobileViewTestCase, cls).setUpClass()
+        cls.none_mobile_course = ToyCourseFactory.create()
+        cls.mobile_course = ToyCourseFactory.create(mobile_available=True, course='mobile', run='2018_Spring')
+
+    def setUp(self):
+        super(CompletionMobileViewTestCase, self).setUp()
+        self.test_user = UserFactory.create()
+        CourseEnrollment.enroll(self.test_user, self.mobile_course.id)
+        CourseEnrollment.enroll(self.test_user, self.none_mobile_course.id)
+        self.mobile_client = APIClient()
+        self.mobile_client.force_authenticate(user=self.test_user)
+
+    @ddt.data(
+        (
+            1,
+            [
+                {
+                    'course_key': 'edX/mobile/2018_Spring',
+                    'completion': {
+                        'earned': 1,
+                        'possible': 0,
+                        'ratio': 0,
+                    },
+                }
+            ]
+        ),
+        (
+            2,
+            [
+                {
+                    'course_key': 'edX/mobile/2018_Spring',
+                    'completion': {
+                        'earned': 1,
+                        'possible': 0,
+                        'ratio': 0,
+                    },
+                },
+                {
+                    'course_key': 'edX/toy/2012_Fall',
+                    'completion': {
+                        'earned': 0,
+                        'possible': 12.0,
+                        'ratio': 0,
+                    },
+                }
+            ]
+        )
+    )
+    @ddt.unpack
+    def test_list_view_mobile_only(self, expected_count, expected_results):
```

Comment:
This needs something to make the request different between the two test variants.  Another ddt arg like `{'mobile_only': 'true'}` or something?

Reply:
You're right, in a way. I should have set the value to 1 instead of 2 for the second. (For the record I was using github to run the unit tests because running it locally just doesn't work).